src/highdicom/legacy/sop.py

```diff
@@ -25,384 +31,506 @@
-            # PET Image Frame Type (M)
-            perframe_item.PETImageFrameTypeSequence = [
-                frame_type_item,
-            ]
-
-        # Frame VOI LUT (U)
-        try:
-            frame_voi_lut_item = Dataset()
-            frame_voi_lut_item.WindowCenter = ds.WindowCenter
-            frame_voi_lut_item.WindowWidth = ds.WindowWidth
-            perframe_item.FrameVOILUTSequence = [
-                frame_voi_lut_item,
-            ]
-        except AttributeError:
-            pass
-
-        # Referenced Image (C)
-        try:
-            perframe_item.ReferencedImageSequence = \
-                ds.ReferencedImageSequence
-        except AttributeError:
-            pass
+            v11 = v1
+            v22 = v2
+        if len(v11) != len(v22):
+            return False
+        for xx, yy in zip(v11, v22):
+            if isinstance(xx, DSfloat) or isinstance(xx, float):
+                if not is_equal_float(xx, yy):
+                    return False
+            else:
+                if xx != yy:
+                    return False
+        return True
 
-        # Derivation Image (C)
-        try:
-            perframe_item.SourceImageSequence = ds.SourceImageSequence
-        except AttributeError:
-            pass
+    @classmethod
+    def isequal_dicom_dataset(cls, ds1: Dataset, ds2: Dataset) -> bool:
+        """Checks if two dicom dataset have the same value in all attributes
+        Parameters
+        ----------
+        ds1: Dataset
+            1st dicom dataset
+        ds2: Dataset
+            2nd dicom dataset
+        Returns
+        -------
+        True of dicom datasets are equal otherwise False
+        """
+        if type(ds1) != type(ds2):
+            return False
+        if not isinstance(ds1, Dataset):
+            return False
+        for k1, elem1 in ds1.items():
+            if k1 not in ds2:
+                return False
+            elem2 = ds2[k1]
+            if not cls.isequal(elem2.value, elem1.value):
+                return False
+        return True
+
+    @classmethod
+    def tag2str(cls, tg: BaseTag) -> str:
+        """Converts tag to hex format like (group, element)
 
-        # Frame Anatomy (C)
-        try:
-            frame_anatomy_item = Dataset()
-            frame_anatomy_item.BodyPartExamined = ds.BodyPartExamined
-            perframe_item.FrameAnatomySequence = [
-                frame_anatomy_item,
-            ]
-        except AttributeError:
-            pass
-
-        # Image Frame Conversion Source (C)
-        conv_src_attr_item = Dataset()
-        conv_src_attr_item.ReferencedSOPClassUID = ds.SOPClassUID
-        conv_src_attr_item.ReferencedSOPInstanceUID = ds.SOPInstanceUID
-        perframe_item.ConversionSourceAttributesSequence = [
-            conv_src_attr_item,
-        ]
+        """
+        if not isinstance(tg, BaseTag):
+            tg = Tag(tg)
+        return '(0x{:0>4x}, 0x{:0>4x})'.format(tg.group, tg.element)
```

Comment:
See [pydicom.tag.BaseTag.__str__](https://github.com/pydicom/pydicom/blob/master/pydicom/tag.py#L193)

Reply:
Sounds trivial, but I wanted to include '0x' for hex representation :). Don't get why they didn't include hex signe.

Reply:
Removed that method.